server/app/src/main/java/com/google/heartrate/wearos/app/bluetooth/server/BluetoothAdvertiser.java

```diff
@@ -0,0 +1,87 @@
+package com.google.heartrate.wearos.app.bluetooth.server;
+
+import android.bluetooth.le.AdvertiseCallback;
+import android.bluetooth.le.AdvertiseData;
+import android.bluetooth.le.AdvertiseSettings;
+import android.bluetooth.le.BluetoothLeAdvertiser;
+import android.content.Context;
+import android.os.ParcelUuid;
+import android.util.Log;
+
+import com.google.heartrate.wearos.app.bluetooth.BluetoothUtils;
+import com.google.heartrate.wearos.app.gatt.GattException;
+
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * {@link BluetoothAdvertiser} is the wrapper over {@link BluetoothLeAdvertiser} class.
+ * <p>
+ * Advertising allows devices to broadcast information defining their intentions.
+ * Server uses advertising to allow clients to find it and, subsequently, to establish a connection with it.
+ * <p>
+ * A server can use {@link BluetoothAdvertiser} to advertise its existence and show it's complete or partial
+ * list of GATT services it offers.
+ *
+ */
+public class BluetoothAdvertiser {
+    private static final String TAG = BluetoothAdvertiser.class.getSimpleName();
+
+    /** {@link BluetoothAdvertiser} for advertising process. */
+    private final BluetoothLeAdvertiser bluetoothLeAdvertiser;
+
+    /** {@link AdvertiseCallback} to receive async responses from {@link BluetoothAdvertiser}. */
+    final AdvertiseCallback advertiseCallback = new AdvertiseCallback() {
+        @Override
+        public void onStartSuccess(AdvertiseSettings settingsInEffect) {
+            Log.i(TAG, "BluetoothLE Advertise Started.");
+        }
+
+        @Override
+        public void onStartFailure(int errorCode) {
+            Log.e(TAG, String.format("BluetoothLE Advertise Failed: %d",  + errorCode));
+        }
+    };
+
+    public BluetoothAdvertiser(Context context) throws GattException {
+        bluetoothLeAdvertiser = BluetoothUtils.getBluetoothLeAdvertiser(context);
+    }
+
+    /**
+     * Start advertising to clients about server existence.
+     *
+     * @param serviceUuids list of GATT services server offers
+     */
+    public void startAdvertisingServices(Set<UUID> serviceUuids) {
```

Comment:
nit: Do you need a set? You don't use any of the set operations. I think an iterable or list would be enough here.